en/ros/ros2_offboard_control.md

```diff
@@ -0,0 +1,193 @@
+ *        vehicle hover at 5 meters.
+ */
+void OffboardControl::publish_position_setpoint_triplet() const {
+	PositionSetpointTriplet msg{};
+	msg.timestamp = timestamp_.load();
+	msg.current.timestamp = timestamp_.load();
+	msg.current.type = PositionSetpoint::SETPOINT_TYPE_POSITION;
+	msg.current.x = 0.0;
+	msg.current.y = 0.0;
+	msg.current.z = -5.0;
+	msg.current.cruising_speed = -1.0;
+	msg.current.position_valid = true;
+	msg.current.alt_valid = true;
+	msg.current.valid = true;
+
+	position_setpoint_triplet_publisher_->publish(msg);
+}
+```
+
+The above functions exemplify how the fields on both `offboard_control_mode` and `position_setpoint_triplet`
+messages can be set. Notice that the above example is applicable for offboard position control,
+where on the `offboard_control_mode` message, the `ignore_position` field is set to `true`, while the all
+the others get set to false, and in the `position_setpoint_triplet`, on the `current` setpoint,
+`valid`, `alt_valid` and `position_valid` are set to true. Also, in this case, `x`, `y` and `z` fields
+are hardcoded to certain values, but they can be updated dynamically according to an algorithm or even
+by a subscription callback for messages coming from another node.
+
+:::tip
+The position is already being published in the NED coordinate frame for simplicity, but in the case of
+the user wanting to subscribe to data coming from other nodes, and since the standard frame of reference in
+ROS/ROS2 is ENU, the user can use the available helper functions in the [`frame_transform` library](https://github.com/PX4/px4_ros_com/blob/master/src/lib/frame_transforms.cpp).
+:::
+
+```c++
+/**
+ * @brief Publish vehicle commands
+ * @param command   Command code (matches VehicleCommand and MAVLink MAV_CMD codes)
+ * @param param1    Command parameter 1
+ * @param param2    Command parameter 2
+ */
+void OffboardControl::publish_vehicle_command(uint16_t command, float param1,
+					      float param2) const {
+	VehicleCommand msg{};
+	msg.timestamp = timestamp_.load();
+	msg.param1 = param1;
+	msg.param2 = param2;
+	msg.command = command;
+	msg.target_system = 1;
+	msg.target_component = 1;
+	msg.source_system = 1;
+	msg.source_component = 1;
+	msg.from_external = true;
+
+	vehicle_command_publisher_->publish(msg);
+}
+```
+
+As the description suggests, the above code serves the purpose of sending `vehicle_command_publisher`
+messages with commands to the flight controller.
+
+:::note
+By the time of writing, `vehicle_command_publisher` is also already configured to be received.
+:::
+
+## Usage
+
+After building the colcon workspace, and after starting PX4 SITL and both the microRTPS bridge client and agent:
+
+```sh
+$ source path_to_colcon_workspace/install/setup.bash
+$ ros2 run px4_ros_com offboard_control
+```
+
+## Demo with PX4 SITL and Gazebo
+
+<video class="video-stream html5-main-video" controls="controls" src="blob:https://www.youtube.com/db7723cd-2b7e-4612-92cc-02fa710abde9">
```

Comment:
Have you rendered all this in Vuepress?

I mean generally, and also this particular line/video. In Vuepress you can do video like `![@youtube](this-is-share-url)`

Reply:
Ah I didn't know this was using Vuepress. Have to check that.